packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CopyFilesControls.tsx

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+
+import { ViewElement } from '../../context/elements';
+
+import { Controls } from '../Controls';
+
+import { Title } from './Controls/Title';
+import { displayText } from '../../displayText/en';
+import { CLASS_BASE } from '../constants';
+import { DestinationPicker } from './DestinationPicker';
+
+import { useCopyView } from './CopyView/useCopyView';
+import { DataTableControl } from '../../controls/DataTableControl';
+import { ControlsContextProvider } from '../../controls/context';
+import { getActionViewTableData } from './utils';
+import { useStore } from '../../providers/store';
+import { ControlsContext } from '../../controls/types';
+import { ActionStartControl } from '../../controls/ActionStartControl';
+import { getTasksHaveStarted } from './utils';
+import { DescriptionList } from '../../components/DescriptionList';
+import { StatusDisplayControl } from '../../controls/StatusDisplayControl';
+import { getDestinationListFullPrefix } from './utils/getDestinationPickerDataTable';
+import { ActionCancelControl } from '../../controls/ActionCancelControl';
+import { CopyHandlerData } from '../../actions';
+
+const { Exit } = Controls;
+const { actionSetDestination } = displayText;
+
+export const CopyFilesControls = ({
+  onExit: _onExit,
+}: {
+  onExit?: () => void;
+}): React.JSX.Element => {
+  const {
+    destinationList,
+    onSetDestinationList,
+    disableCancel,
+    disableClose,
+    disablePrimary,
+    onExit,
+    onActionCancel,
+    onActionStart,
+    taskCounts,
+    tasks,
+  } = useCopyView({ onExit: _onExit });
+
+  const [{ location }] = useStore();
+  const { current } = location;
+  const path = current?.prefix;
+  const tableData = getActionViewTableData<CopyHandlerData>({
+    tasks,
+    taskCounts,
+    path: path ?? '',
+  });
+
+  const contextValue: ControlsContext = {
+    data: {
+      taskCounts,
+      tableData,
+      actionStartLabel: 'Start',
+      isActionStartDisabled: disablePrimary,
+      isActionCancelDisabled: disableCancel,
+      actionCancelLabel: 'Cancel',
+    },
+    actionsConfig: { type: 'BATCH_ACTION', isCancelable: true },
+    onActionStart,
+    onActionCancel,
+  };
+  const hasStarted = getTasksHaveStarted(taskCounts);
```

Comment:
This can just be replaced by exposing `isProcessing` from `useProcessTasks` on `useCopyView` (follow up)